61 update docs about workflow (#62)
xpirit-training/demo-features-issueops: README.md

# Demo Repository for IssueOps

Project demonstrating the power of issue based process handling using github issues and actions.

## Issue Templates

### Repository Management

#### Repository Request: [repo-request.yml](.github/ISSUE_TEMPLATE/repo-request.yml)

Issue provising several inputs and dropdown selections to collect data required for repository creation via [workflow](#handle-repository-request-handle-repo-requestyml).

The dropdown selections are automatically updated by the [repo request update](#update-repo-request-issue-template-update-repo-request-templateyml) workflow.

#### Repository Deletion: [repo-deletion.yml](.github/ISSUE_TEMPLATE/repo-deletion.yml)

> Simple issue form just collection the name of the repo which shall be deleted via [workflow](#handle-repository-deletion-handle-repo-deletionyml).

Reviewer: ```suggestion
Simple issue form taking the name of the repository as input which is supposed to be deleted via [workflow](#handle-repository-deletion-handle-repo-deletionyml).
```

## Workflows

### Issue Handling

Issues are handled in a three-tiered workflow.

1. Select the right workflow based on an issue label
1. Handle the issue and its data
1. Handle the corresponding task

```mermaid
flowchart TD    
    A(Issue Labeled) --> B(Handle Issue)
    style A fill:#d6d6d6,stroke:#000000,stroke-dasharray: 5 5
    B -->|"[label=repo-request]"| C(Handle Repository Request Issue)
    B -->|"[label=repo-deletion]"| D(Handle Repository Deletion Issue)
    C -->|repo attributes| E(Create Repository)
    D -->|repo name| F(Delete Repository)
    E --> G(Repository Created)
    style G fill:#d6d6d6,stroke:#000000,stroke-dasharray: 5 5
    F --> H(Repository Deleted)
    style H fill:#d6d6d6,stroke:#000000,stroke-dasharray: 5 5
```

#### Handle Issue: [handle-issue.yml](.github/workflows/handle-issue.yml)

Workflow handling the [issue templates](#issue-templates).
Depending on the given [label](https://docs.github.com/issues/using-labels-and-milestones-to-track-work/managing-labels) the workflow executes the corresponding [subworkflow](https://docs.github.com/actions/using-workflows/reusing-workflows).

The following subworkflows may be triggered:

> - [repository request](#handle-repository-request-handle-repo-requestyml)
> - [repository deletion](#handle-repository-deletion-handle-repo-deletionyml)

Reviewer: ```suggestion
- [Repository request](#handle-repository-request-handle-repo-requestyml)
- [Repository deletion](#handle-repository-deletion-handle-repo-deletionyml)
```

#### Delete Repository: [repo-deletion.yml](.github/workflows/repo-deletion.yml)

Workflow deleting a repository by name.

### Organization Management

#### Update Repo Request Issue Template: [update-repo-request-template.yml](.github/workflows/update-repo-request-template.yml)

Workflow that is triggered by any changes to the workflow file `.github/workflows/update-repo-request-template.yml` òr the `./org` or `./templates/languages` directories. The workflow reads the contents from the folder expecting the following structure:

```
.
├── org/
│   ├── company-1/
│   │   ├── metadata.yml
│   │   └── teams/
│   │       └── team-1/
│   │           ├── metadata.yml
│   │           └── projects/
│   │               ├── project-1/
│   │               └── project-2/
│   └── company-n/
│       ├── metadata.yml
│       └── teams/
│           └── team-n/
│               └── metadata.yml/
│                   └── projects/
│                       └── project-n/
└── templates/
    └── languages/
        ├── language-1/
        │   ├── README.md
        │   └── language-files
        └── language-n/
            ├── README.md
            └── language-files
```

> The `org` is defined of a set of `company` which has a set of `teams`. Each team has a set of `projects`. In order to define so general properties on `company` as well as `team` level, each of those has a `metadata.yml` This file holds properties such as team manager oder team identifier.

Reviewer: ```suggestion
The `org` is defined of a set of `company` which has a set of `teams`. Each team has a set of `projects`. In order to define the general properties on `company` as well as `team` level, each of those has a `metadata.yml`. This file holds properties such as team manager or team identifier.
```

Any change to the structure will trigger the workflow and populate the issue template.

## Actions

### Issue Handling

> #### [comment-issue](.github/actions/comment-issue/action.yml)

Author: done
